File: Exercicios/Java/Lista1/src/Ex5.java
import java.io.IOException;

/**
 * SSC0103 - Programação Orientada a Objetos
 * Lista de exercícios #1 (Java)
 * Exercício 05
 * 
 * @author Ana Lívia Ruegger Saldanha (N.USP 8586691)
 *
 */
public class Ex5 {
	
	/**
	 * Este método lê um número inteiro, verifica se ele é primo e,
	 * caso ele não seja, exibe qual o seu menor divisor.
	 * 
	 * @param args
	 * @throws IOException
	 */
	public static void main(String[] args) throws IOException {
		int n = EntradaTeclado.leInt();
		for (int i = 2; i <= n/2; i++) {
			if (n % i == 0) {
				System.out.printf("O número %d é divisível por %d.\n", n, i);
				return;
			}
		}
		System.out.printf("O número %d é primo.\n", n);
	}

}

File: Exercicios/Python/Lista1/Ex5.py
# SSC0103 - Programação Orientada a Objetos
# Lista de exercícios #1 (Python)
# Exercício 05
# 
# Ana Lívia Ruegger Saldanha (N.USP 8586691)

'''
Este programa lê um número inteiro, verifica se ele é primo e, caso ele
não seja, exibe qual o seu menor divisor.
'''


def is_prime(n):
    for i in range(2, int(n/2 + 1)):
        if n % i == 0:
            print("O número", n, "é divisível por", i)
            return
    print("O número", n, "é primo")

n = int(input())
is_prime(n)